camelot/parsers/base.py

```diff
@@ -1,17 +1,21 @@
 # -*- coding: utf-8 -*-
 
 import os
+from typing import Dict, Optional, Any
 
 from ..utils import get_page_layout, get_image_and_text_objects
 
 
 class BaseParser(object):
     """Defines a base parser."""
 
-    def _generate_layout(self, filename, layout_kwargs):
+    def _generate_layout(self, filename, layout_kwargs, page_layout_per_page: Optional[Dict[str, Any]] = None):
         self.filename = filename
         self.layout_kwargs = layout_kwargs
-        self.layout, self.dimensions = get_page_layout(filename, **layout_kwargs)
+        if page_layout_per_page is None:
+            self.layout, self.dimensions = get_page_layout(filename, **layout_kwargs)
+        else:
+            self.layout, self.dimensions = page_layout_per_page["layout"], page_layout_per_page["dimensions"]
```

Comment:
Changed so that the results of the following 1st get_page_layout can be reused here.
https://github.com/nexus-lib/camelot/blob/feature/improve_process_speed/camelot/handlers.py#L122